cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -1182,22 +1184,25 @@ class AsofJoinNode : public ExecNode {
 
       for (int i = 0; i < input_schema[j]->num_fields(); ++i) {
         const auto field = input_schema[j]->field(i);
+        bool as_output;
```

Comment:
```suggestion
        bool as_output = false;
```

This might be what is causing the Windows CI segfaults.

Reply:
That would be a great catch (and perhaps a compiler bug, since `as_output` is initialized in every code path?). I'll try this.

Reply:
Ah, you're right.  Probably not it then.

Reply:
What is this variable supposed to indicate? Something like 
```
Indicate whether the input column is part of output columns. This is false for "on" and "by" columns from right tables and true for all other columns
```